### PR TITLE
using UPMEM_PROFILE env variables to determine DPU target

### PR DESCRIPTION
**Pull request by JordiChauzi**
cf upmem/dpu_tools#27

### CODE REVIEW

File: source/Plugins/Process/Dpu/DpuRank.cpp [hunk 1]
```diff
@@ -78,10 +56,19 @@ bool DpuRank::Open(llvm::StringRef profile) {
   } else if (args_split.first.equals("simulator")) {
     m_type = FUNCTIONAL_SIMULATOR;
   } else {
-    m_type = GetTypeFromDpuDiagnostic();
+    m_type = HW;
```
Comment (vpalatin) [inline source/Plugins/Process/Dpu/DpuRank.cpp]: in the current state of thing, this is mildly satisfying.
Personally I´m often running:
dpu-lldb my_dpu_binary.bin

having HW as default is a long term goal, but we likely need a fallback to simulator ?

File: source/Plugins/Process/Dpu/DpuRank.cpp [hunk 2]
```diff
@@ -78,10 +56,19 @@ bool DpuRank::Open(llvm::StringRef profile) {
   } else if (args_split.first.equals("simulator")) {
```
Comment (vpalatin) [inline source/Plugins/Process/Dpu/DpuRank.cpp]: do we still the  args_split = profile.split('/') above ?
we need to change this syntax too ?

Comment (JordiChauzi) [reply source/Plugins/Process/Dpu/DpuRank.cpp]: We may want to change that. I don't know what @rjodinupmem thinks about it.

File: source/Plugins/Process/Dpu/DpuRank.cpp (hunk 1 above)
Comment (JordiChauzi) [reply source/Plugins/Process/Dpu/DpuRank.cpp]: I mean, we could try to implement something that check if we have any hardware before choosing the default (I am not sure we have everything we need for this in the `hw` backend at the moment). But isn't it why we wanted `UPMEM_PROFILE_BASE` ? To set user specific default properties ?

**Pull request merged**